src/data/polynomial/splits.lean

```diff
@@ -311,6 +311,17 @@ begin
   simp,
 end
 
+theorem coeff_mem_subring_of_splits {f : K[X]} (hs : f.splits (ring_hom.id K))
```

Comment:
Can you replace `T : subring K` by `R` with `[comm_ring R] [algebra R K]`? The condition `hr` should be that the roots are in the image of `algebra_map`. Also you can use [polynomial.lifts](https://leanprover-community.github.io/mathlib_docs/data/polynomial/lifts.html#polynomial.lifts) to say that `f` has coefficients in `R`.